automations/js/src/utils/projects.mjs

```diff
@@ -0,0 +1,229 @@
+        contentId: issueId,
+      }
+    )
+    const card = res.addProjectV2ItemById.item
+    return {
+      id: card.id,
+      status: card.fieldValueByName?.name, // `null` if new card
+    }
+  }
+
+  /**
+   * Set the value of the custom choice field to the given option.
+   *
+   * @param cardId {string} the ID of the card whose field is to be updated
+   * @param fieldName {string} the name of the field to update
+   * @param optionName {string} the updated value of the field
+   * @returns {Promise<string>} the ID of the card that was updated
+   */
+  async setCustomChoiceField(cardId, fieldName, optionName) {
+    // Preliminary validation
+    if (!this.fields[fieldName]) {
+      throw new Error(`Unknown field name "${fieldName}".`)
+    }
+    if (!this.fields[fieldName].options[optionName]) {
+      throw new Error(
+        `Unknown option name "${optionName}" for field "${fieldName}".`
+      )
+    }
+
+    const res = await this.octokit.graphql(
+      `mutation setCustomField($projectId: ID!, $itemId: ID!, $fieldId: ID!, $optionId: String!) {
+        updateProjectV2ItemFieldValue(input: {
+          projectId: $projectId,
+          itemId: $itemId,
+          fieldId: $fieldId,
+          value: {singleSelectOptionId: $optionId}
+        }) {
+          projectV2Item {
+            id
+          }
+        }
+      }`,
+      {
+        projectId: this.projectId,
+        itemId: cardId,
+        fieldId: this.fields[fieldName].id,
+        optionId: this.fields[fieldName].options[optionName],
+      }
+    )
+    return res.updateProjectV2ItemFieldValue.projectV2Item.id
+  }
+
+  /**
+   * Moving a card to a different column is akin to updating the custom field
+   * "Status" that is present on all GitHub project boards.
+   *
+   * @param cardId {string} the ID of the card to move
+   * @param destColumn {string} the name of the column where to move it
+   * @returns {Promise<string>} the ID of the card that was moved
+   */
+  async moveCard(cardId, destColumn) {
+    return await this.setCustomChoiceField(cardId, 'Status', destColumn)
+  }
+}
+
+/**
+ * Get the `Project` instance for the project board with the given name.
+ *
+ * @param name {string} the name of the project (without the 'Openverse' prefix)
+ * @returns {Project} the `Project` instance to interact with the project board
+ */
+export async function getBoard(name) {
+  const octokit = getOctokit()
+
+  let project
+  switch (name) {
+    case 'Backlog':
+      project = new Project(octokit, 'WordPress', 75)
+      break
+    case 'PRs':
+      project = new Project(octokit, 'WordPress', 98)
+      break
+    case 'Todos':
+      project = new Project(octokit, 'WordPress', 59)
+      break
+    case 'Discussions':
+      project = new Project(octokit, 'WordPress', 79)
+      break
+    case 'Project Tracker':
+      project = new Project(octokit, 'WordPress', 70)
+      break
+    default:
+      throw new Error(`Unknown project board "${name}".`)
+  }
```

Comment:
nit: I think this can be significantly simplified and made easier to update by using an object to configure the mapping of project name to id:

```suggestion
  const projects = {
    Backlog: 75,
    PRs: 98,
    Todos: 59,
    Discussions: 79,
    "Project Tracker": 70,
  }
  
  if (!projects[name]) {
    throw new Error(`Unknown project board "${name}".`)
  }
  
  const project = new Project(octokit, 'WordPress', projects[name])
```

This makes adding/removing projects a bit easier, just one line to add instead of 3, and makes the relationship of each case statement clearer (it's really just 1-to-1 project name to project id).

Reply:
This is really great, and should've been the first solution I thought of. I don't even remember why I used `switch`-`case` in the first place 🤦. Fixed in aaf807f.